### DISCUSSION TITLE
The plan of upgrading v6d to support C++17

### DISCUSSION THREAD

**Discussion by songqing** [v6d-io/v6d#1146] [General]
As Arrow has been upgraded to support C++17, and more new features have been introduced in C++17, so, I want to know what's the plan of v6d to do this.
Thank you.

**Comment by sighingnow** [comment]
Vineyard currently can be built with arrow 10 with C++ compilers that supports C++17, vineyard itself has no plan to disabling the support for C++14 -only compilers.

**Comment by sighingnow** [comment]
Is there any issues that vineyard cannot be built with arrow 10 in your environment?

**Comment by sighingnow** [comment]
By the way vineyard can be built with arrow>=6.0.0.

**Comment by songqing** [comment]
ok, I see. Arrow's build is ok for me, I want to do some development with v6d, however, some of my code is developed with C++17, and it may be difficult to integrate into v6d with c++14, so, if v6d upgrade to support c++17, it will be easy for me to develop.

**Comment by sighingnow** [comment]
It would be great if you could submit a work-in-progress PR first then we can review the changes to see whether C++17 is a hard requirements in detail.

If C++17 is strong requirements we definitely can upgrade to arrow>=10 and C++17. We current keep with c++14 just because we see no reason for adding extra constraints, i.e., keeping C++14 is prefered, but not a hard requirement.

**Comment by songqing** [comment]
OK, I see, I'll try first, thank you.

**Comment by sighingnow** [comment]
Closing as discussed. Thanks!
